config/config.go

```diff
@@ -588,6 +588,7 @@ type RemoteWriteConfig struct {
 	Headers             map[string]string `yaml:"headers,omitempty"`
 	WriteRelabelConfigs []*relabel.Config `yaml:"write_relabel_configs,omitempty"`
 	Name                string            `yaml:"name,omitempty"`
+	RetryOnRateLimit    bool              `yaml:"retry_on_http_429,omitempty"`
```

Comment:
I agree with @roidelapluie, this should be a config under `QueueConfig` next to backoff configurations.